backend/hitas/models/condition_of_sale.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+from typing import Optional
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+from enumfields import Enum, EnumField
+
+from hitas.models._base import ExternalHitasModel
+
+
+class GracePeriod(Enum):
+    NOT_GIVEN = "not_given"
+    THREE_MONTHS = "three_months"
+    SIX_MONTHS = "six_months"
+
+    class Labels:
+        NOT_GIVEN = _("Not given")
+        THREE_MONTHS = _("Three months")
+        SIX_MONTHS = _("Six months")
+
+
+# 'Myyntiehto'
+class ConditionOfSale(ExternalHitasModel):
+    new_ownership = models.ForeignKey(
+        "Ownership",
+        related_name="conditions_of_sale_new",
+        on_delete=models.CASCADE,
+        editable=False,
+    )
+    old_ownership = models.ForeignKey(
+        "Ownership",
+        related_name="conditions_of_sale_old",
+        on_delete=models.CASCADE,
+        editable=False,
+    )
```

Comment:
I think in this case the ownership of the old apartment should be corrected so that it reflects the actual owners of that apartment. Then we can create the condition of sale for the correct ownerships, since Person A does not actually have to give up their share of the ownership, only Person B's share, which is would be falsely included in Person A's share.

Reply:
In my example **Person A** does need to give up their share of the old apartment, because **Person A** and **B** are in the same household (and also will be in the future).
In this example **Person B** also does not own any shares in the old apartment, and **Person A** will not be owning any shares of the new apartment.

Reply:
I thought that people sharing a household would implicitly share the shares for that apartment? That is how we have been talking about it with the customer at least, that the "other person needs to give up their share".

In a shared household this will usually require the other person to also I've up their share, since they likely want to continue sharing the household, but that is just one possible arrangement between the owners.

Reply:
> would implicitly share the shares for that apartment
No, owning shares is personal. The need for the other person to give up shares comes from being in the household.

Two people owning shares in an apartment does not mean they belong in the same household, and also two people can belong in the same household without both of them owning shares in the apartment they live in.

Reply:
Ok, then we should remove the restriction. To support these cases, I think we should include an optional field in the condition of sale create endpoint under owner where you can add additional ownerships that are for new apartments the owner doesn't have ownerships in. We can then create additional conditions of sale between these ownerships and the persons current ownership.

Reply:
Or, if we replace `new_ownership` with `new_apartment` we remove one more 'duplicate' field (`ownership.owner`) and remove the need to add a public id to `Ownership`, while keeping every request the same (no need for optional fields) 😉

Reply:
Using just an apartment reference will make canceling the sale (or removing the ownership for the new apartment some other way if we decide so in the future) for the new apartment not automatically fulfill its conditions of sale, so I would prefer using two ownership references and leave it to cascade.

Reply:
Added the option to add additional ownerships for which to create conditions of sale. This brings back the ownership UUID so that it can be used to set the additional ownerships.

Reply:
Ownership UUID is now removed since the new API uses households with owner UUIDs.